CMakeLists.txt

```diff
@@ -319,69 +346,86 @@ add_definitions(-D__STDC_LIMIT_MACROS)
 add_definitions(-D__STDC_CONSTANT_MACROS)
 
 ###### Test Inputs ######
-if(GRAPH_LOCATION AND EXISTS "${GRAPH_LOCATION}")
-  MESSAGE(STATUS "USING graph location ${GRAPH_LOCATION}")
-  set(BASEINPUT "${GRAPH_LOCATION}")
-elseif(EXISTS /net/ohm/export/iss/inputs)
-  set(BASEINPUT /net/ohm/export/iss/inputs)
-  MESSAGE(STATUS "Using graph location /net/ohm/export/iss/inputs")
+if(GRAPH_LOCATION)
+  MESSAGE(STATUS "Using graph location ${GRAPH_LOCATION}")
+  set(BASEINPUT "${GRAPH_LOCATION}/small_inputs")
 else()
   add_subdirectory(inputs)
   set(BASEINPUT ${CMAKE_BINARY_DIR}/inputs/small_inputs)
-  MESSAGE(STATUS "use make input to download inputs in the build directory")
-endif()
-
-###### Test Outputs ######
-if(EXISTS /net/ohm/export/iss/dist-outputs)
-  set(BASEOUTPUT /net/ohm/export/iss/dist-outputs)
-elseif(EXISTS /net/ohm/export/cdgc/dist-outputs)
-  set(BASEOUTPUT /net/ohm/export/cdgc/dist-outputs)
-elseif(EXISTS /workspace/dist-outputs)
-  set(BASEOUTPUT /workspace/dist-outputs)
+  MESSAGE(STATUS "Use 'make input' to download inputs in the build directory")
 endif()
 
 ###### Global Functions ######
 
-function(add_internal_library name)
-  add_library(${name} OBJECT ${ARGN})
-endfunction()
-
 function(add_test_scale type app)
-  if(NUM_TEST_THREADS)
-    set(N ${NUM_TEST_THREADS})
-  else()
-    cmake_host_system_information(RESULT N QUERY NUMBER_OF_PHYSICAL_CORES)
-  endif()
-  if (N EQUAL 0)
-    set(N 1)
-  endif()
-  if (N LESS 8)
-    foreach(thr RANGE 1 ${N})
-      add_test(test-${type}-${app}-${thr} ${app} ${ARGN} -t ${thr})
-    endforeach(thr)
-  else()
-    foreach(part RANGE 1 8)
-      math(EXPR thr "(${N} * ${part})/ 8")
-      if ((part EQUAL 1) AND NOT (thr EQUAL 1))
-        add_test(test-${type}-${app}-1 ${app} ${ARGN} -t 1)
-      endif()
-      add_test(test-${type}-${app}-${thr} ${app} ${ARGN} -t ${thr})
-    endforeach(part)
+  set(options NOT_QUICK)
+  set(one_value_args)
+  set(multi_value_args REQUIRES COMMAND_PREFIX)
+  cmake_parse_arguments(X "${options}" "${one_value_args}" "${multi_value_args}" ${ARGN})
+
+  set(threads)
+  set(thr 1)
+  while (${thr} LESS ${NUM_TEST_THREADS})
```

Comment:
I have been thinking of these run-tests as more like holistic spot checks (for functional correctness), so good coverage would be "one threads", "some threads", "lots of threads" (e.g,, black box versus white box testing).

(I'd actually be happy to move from powers-of-two to just 1, n/2, and n.)

If there have been cases of issues related to the treatment of packages, I can see some value of being more specific, but then I'd push for a good unit test, independent of any app, first.

In general, I'm always a little wary of being too machine specific here because it can become a source of confusion for developers and maintenance burden down the line.

If these are intended to be performance tests, then I certainly agree with you.

I don't mind going back to multiples of 8 but I would probably also separate the more exhaustive testing from the quick tests and have a plan to run more aggressive testing out-of-line from PR integration tests.

Reply:
Good point. Why not do it reverse? Keep diving from max by 2 until 1, but run from 1 till max in that list.

Reply:
We've looked at running more aggressive performance regression tests as well, but those plans have been sidelined for the time being. +1 for the "go down by powers of 2" idea.